packages/boot/src/__tests__/acceptance/configuration.booter.acceptance.ts

```diff
@@ -0,0 +1,173 @@
+// Copyright IBM Corp. 2019,2020. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  Application,
+  ContextTags,
+  createBindingFromClass,
+  TagValueMatcher,
+} from '@loopback/core';
+import {expect, TestSandbox} from '@loopback/testlab';
+import {Suite} from 'mocha';
+import {resolve} from 'path';
+import {ConfigurationLoader, configurationLoader} from '../../booters';
+import {BooterApp} from '../fixtures/application';
+
+describe('configuration booter acceptance tests', function (this: Suite) {
+  this.timeout(5000);
+  let app: BooterApp;
+  const sandbox = new TestSandbox(resolve(__dirname, '../../.sandbox'), {
+    // We intentionally use this flag so that `dist/application.js` can keep
+    // its relative path to satisfy import statements
+    subdir: false,
+  });
+
+  beforeEach('reset sandbox', () => sandbox.reset());
+  beforeEach(getApp);
+
+  it('binds configuration objects for js', async () => {
+    await copyJs();
+    await testConfigBooting('js');
+  });
```

Comment:
I find these tests super difficult to understand. One has to dig deep into several other functions and fixture files to understand what is being tested and asserted. Please find a way how to better show the intent of the tests. 

IMO, these tests are also very rigidly enforcing the same set of config entries (keys and values). When we need to test a different config use case in the future, we will be able to reuse very little and will have to write most of the code from scratch.

An example illustrating what I consider as a better solution:

```ts
it('binds configuration objects for js', async () => {
  await fs.writeFileSync('dist/configs/test.config.js', `
export = {
   'loggers.Log1': {
     level: 'info',
     _source: 'js',
   },
   'loggers.Log2': {
     level: 'debug',
     _source: 'js',
   },
 };
`);

  await app.boot();

  expect(await app.getConfig('loggers.Log1')).to.eql({level: 'info', _source: source});
  expect(await app.getConfig('loggers.Log2')).to.eql({level: 'debug', _source: source});
});
```

Reply:
Agreed, this could be simpler and easier to read.

I personally prefer test cases to be verbose and self-contained than being normalized and "optimized". Test cases are read/revisited as individuals, having the whole context in the test case block helps a lot. Infrastructural helper functions are OK, assertion helper functions should be avoided IMO.

Reply:
I moved json/yaml config inline.